File: bindings/php/src/ProvenBfd.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// BFD protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** BfdState matching the Idris2 ABI tags. */
enum BfdState: int
{
    case AdminDown = 0;
    case Down = 1;
    case Init = 2;
    case Up = 3;
}

/** Diagnostic matching the Idris2 ABI tags. */
enum Diagnostic: int
{
    case NoDiagnostic = 0;
    case ControlDetectionTimeExpired = 1;
    case EchoFunctionFailed = 2;
    case NeighborSignaledSessionDown = 3;
    case ForwardingPlaneReset = 4;
    case PathDown = 5;
    case ConcatenatedPathDown = 6;
    case AdministrativelyDown = 7;
    case ReverseConcatenatedPathDown = 8;
}

/** SessionMode matching the Idris2 ABI tags. */
enum SessionMode: int
{
    case AsyncMode = 0;
    case DemandMode = 1;
}

/** SessionState matching the Idris2 ABI tags. */
enum SessionState: int
{
    case Idle = 0;
    case SsDown = 1;
    case Negotiating = 2;
    case Established = 3;
    case Teardown = 4;
}

File: bindings/csharp/src/ProvenBfd.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// BFD protocol types for proven-servers.

namespace Proven;

/// <summary>BfdState matching the Idris2 ABI tags (0-3).</summary>
public enum BfdState : byte
{
    AdminDown = 0,
    Down = 1,
    Init = 2,
    Up = 3
}

/// <summary>Diagnostic matching the Idris2 ABI tags (0-8).</summary>
public enum Diagnostic : byte
{
    NoDiagnostic = 0,
    ControlDetectionTimeExpired = 1,
    EchoFunctionFailed = 2,
    NeighborSignaledSessionDown = 3,
    ForwardingPlaneReset = 4,
    PathDown = 5,
    ConcatenatedPathDown = 6,
    AdministrativelyDown = 7,
    ReverseConcatenatedPathDown = 8
}

/// <summary>SessionMode matching the Idris2 ABI tags (0-1).</summary>
public enum SessionMode : byte
{
    AsyncMode = 0,
    DemandMode = 1
}

/// <summary>SessionState matching the Idris2 ABI tags (0-4).</summary>
public enum SessionState : byte
{
    Idle = 0,
    SsDown = 1,
    Negotiating = 2,
    Established = 3,
    Teardown = 4
}
